tiledb/sm/query/query.h

```diff
@@ -1190,6 +1189,17 @@ class Query {
    */
   optional<std::string> fragment_name_;
 
+  /** If set, use this timestamp when creating a new fragment. */
+  optional<uint64_t> default_fragment_timestamp_;
+
+  /**
+   * The current timestamp to use for new fragments.
+   *
+   * This is only set once query is initialized or the strategy is created.
+   * Before it is initialized, it is set to nullopt.
+   */
```

Comment:
Depending on if we set the timestamp at init or when creating the strategy, we might be able to get rid of this variable.

Reply:
We can get rid of this if we disable re-initializing the Query when the subarray is set. 

Calling `init` does (and should) over-write a timestamp that was created when the strategy is called. Setting the timestamp in `create_strategy` is necessary for serialization where the strategy is sometimes created before initialization.